src/Provider/AbstractProvider.php

```diff
@@ -414,56 +409,75 @@ protected function sendRequest(RequestInterface $request)
      */
     public function getResponse(RequestInterface $request)
     {
-        $response = (string) $this->sendRequest($request)->getBody();
+        $response = $this->sendRequest($request);
         return $this->parseResponse($response);
     }
 
+
     /**
-     * Get the expected type of response for this provider.
+     * Parses response content as JSON.
      *
-     * @return string
+     * @param string $content JSON content from response body.
+     * @return array Decoded JSON data
+     * @throws UnexpectedValueException if the content could not be parsed.
      */
-    protected function getResponseType()
+    protected function parseJson($content)
     {
-        return static::RESPONSE_TYPE_JSON;
+        $content = json_decode($content, true);
+
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            throw new UnexpectedValueException(sprintf(
+                "Failed to parse JSON response: %s",
+                json_last_error_msg()
```

Comment:
PHP 5.5 allows us to use [`json_last_error_msg()`](http://php.net/manual/en/function.json-last-error-msg.php)